benchmark/bench_dom_api.cpp

```diff
@@ -73,6 +73,79 @@ BENCHMARK(serialize_twitter)->Repetitions(10)->ComputeStatistics("max", [](const
     return *(std::max_element(std::begin(v), std::end(v)));
   })->DisplayAggregatesOnly(true);
 
+
+static void serialize_twitter_to_string(State& state) {
+  dom::parser parser;
+  padded_string docdata;
+  auto error = padded_string::load(TWITTER_JSON).get(docdata);
+  if(error) {
+      cerr << "could not parse twitter.json" << error << endl;
+      return;
+  }
+  // we do not want mem. alloc. in the loop.
+  if((error = parser.allocate(docdata.size()))) {
+      cout << error << endl;
+      return;
+  }
+  dom::element doc;
+  if ((error = parser.parse(docdata).get(doc))) {
+    cerr << "could not parse twitter.json" << error << endl;
+    return;
+  }
+  size_t bytes = 0;
+  for (SIMDJSON_UNUSED auto _ : state) {
+    auto serial = simdjson::to_string(doc);
+    bytes += serial.size();
+    benchmark::DoNotOptimize(serial);
+  }
+  // Gigabyte: https://en.wikipedia.org/wiki/Gigabyte
+  state.counters["Gigabytes"] = benchmark::Counter(
+	        double(bytes), benchmark::Counter::kIsRate,
+	        benchmark::Counter::OneK::kIs1000); // For GiB : kIs1024
+  state.counters["docs"] = Counter(double(state.iterations()), benchmark::Counter::kIsRate);
+}
+BENCHMARK(serialize_twitter_to_string)->Repetitions(10)->ComputeStatistics("max", [](const std::vector<double>& v) -> double {
+    return *(std::max_element(std::begin(v), std::end(v)));
+  })->DisplayAggregatesOnly(true);
+
+static void serialize_twitter_string_builder(State& state) {
+  dom::parser parser;
+  padded_string docdata;
+  auto error = padded_string::load(TWITTER_JSON).get(docdata);
+  if(error) {
+      cerr << "could not parse twitter.json" << error << endl;
+      return;
+  }
+  // we do not want mem. alloc. in the loop.
+  if((error = parser.allocate(docdata.size()))) {
+      cout << error << endl;
+      return;
+  }
+  dom::element doc;
+  if ((error = parser.parse(docdata).get(doc))) {
+    cerr << "could not parse twitter.json" << error << endl;
+    return;
+  }
+  size_t bytes = 0;
+  simdjson::dom::string_builder<> sb;
+  for (SIMDJSON_UNUSED auto _ : state) {
+    sb.clear();
+    sb.append(doc);
+    std::string_view serial = sb.str();
+    bytes += serial.size();
+    benchmark::DoNotOptimize(serial);
+  }
+  // Gigabyte: https://en.wikipedia.org/wiki/Gigabyte
+  state.counters["Gigabytes"] = benchmark::Counter(
+	        double(bytes), benchmark::Counter::kIsRate,
+	        benchmark::Counter::OneK::kIs1000); // For GiB : kIs1024
+  state.counters["docs"] = Counter(double(state.iterations()), benchmark::Counter::kIsRate);
+}
+BENCHMARK(serialize_twitter_string_builder)->Repetitions(10)->ComputeStatistics("max", [](const std::vector<double>& v) -> double {
```

Comment:
I've been meaning to ask, why are you specifying repetitions on things? If you don't specify them, Google Benchmarks will run it until it feels it has a statistically significant result, which seems like a good idea ...

Reply:
>  why are you specifying repetitions on things?  

I think you are not reading the documentation correctly. If you don't specify repetition, then you get one measure. If you want to be able to know what the standard deviation, median, and ... so forth... are... then you need more than one measure.

I'll remove it in this instance, but not because I think it is not useful. It can be really useful to know your margin of error (before you conclude that X is faster than Y, you need to know how wide the fluctuations are).

Reply:
Removed.

Reply:
I see :) When you do want to know this information, you can definitely pass parameters to the executable to make it happen: https://github.com/google/benchmark#statistics-reporting-the-mean-median-and-standard-deviation-of-repeated-benchmarks

```
bench_dom_api --benchmark_repetitions=10 --benchmark_display_aggregates_only=true
```

I think `--benchmark_min_time` might do something similar but cap it to time, which is super useful.